src/sentry/static/sentry/app/stores/globalSelectionStore.jsx

```diff
@@ -34,6 +40,40 @@ const GlobalSelectionStore = Reflux.createStore({
     this.selection = state || getDefaultSelection();
   },
 
+  /**
+   * Initializes the global selection store
+   * If there are query params apply these, otherwise check local storage
+  */
+  loadInitialData(queryParams) {
+    const query = pick(queryParams, Object.values(URL_PARAM));
+    const hasQuery = Object.keys(query).length > 0;
+
+    let globalSelection = getDefaultSelection();
+
+    if (hasQuery) {
+      const parsed = getStateFromQuery(queryParams);
+      globalSelection = {
+        projects: parsed.project || [],
```

Comment:
Do we know if the consumers of global selection will behave well if a user:

1. Has a project stored in local storage
2. They then lose access to one or more stored projects.

Reply:
I'm not sure, but I'll update this to do some validation before putting this into the store